ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/linear_optimisation/algorithms/fillers/CoreProblemFiller.java

```diff
@@ -240,6 +243,38 @@ private void buildRangeActionConstraints(LinearProblem linearProblem) {
         );
     }
 
+    private void updateRangeActionConstraints(LinearProblem linearProblem, RangeActionActivationResult rangeActionActivationResult) {
+        iteration++;
+        optimizationContext.getRangeActionsPerState().entrySet().stream()
+            .sorted(Comparator.comparingInt(e -> e.getKey().getInstant().getOrder()))
+            .forEach(entry ->
+                entry.getValue().forEach(rangeAction -> updateConstraintsForRangeAction(linearProblem, rangeAction, entry.getKey(), rangeActionActivationResult, iteration)
+            )
+        );
+    }
+
+    private void updateConstraintsForRangeAction(LinearProblem linearProblem, RangeAction<?> rangeAction, State state, RangeActionActivationResult rangeActionActivationResult, int iteration) {
+        double previousSetPointValue = rangeActionActivationResult.getOptimizedSetpoint(rangeAction, state);
+        List<Double> minAndMaxAbsoluteAndRelativeSetpoints = getMinAndMaxAbsoluteAndRelativeSetpoints(rangeAction);
+        double minAbsoluteSetpoint = Math.max(minAndMaxAbsoluteAndRelativeSetpoints.get(0), -LinearProblem.infinity());
+        double maxAbsoluteSetpoint = Math.min(minAndMaxAbsoluteAndRelativeSetpoints.get(1), LinearProblem.infinity());
+        double setPointRange = Math.abs(minAbsoluteSetpoint) + Math.abs(maxAbsoluteSetpoint);
+        double constrainedSetPointRange = setPointRange * Math.pow(RANGE_DIMINUTION_RATE, iteration);
+        FaraoMPConstraint iterativeShrink = linearProblem.getRangeActionRelativeSetpointConstraint(rangeAction, state, "iterative_shrink");
+        if (iterativeShrink != null) {
+            iterativeShrink.setLb(previousSetPointValue - constrainedSetPointRange);
+            iterativeShrink.setUb(previousSetPointValue + constrainedSetPointRange);
```

Comment:
shouldn't this have a 0.5 coefficient? 
```suggestion
            iterativeShrink.setLb(previousSetPointValue - 0.5 * constrainedSetPointRange);
            iterativeShrink.setUb(previousSetPointValue + 0.5 * constrainedSetPointRange);
```

Reply:
I do not think we want to introduce a 0.5 coefficient because we chose the factor 2/3 on purpose to allow the PST to come back to its initial tap. If we add this coefficient, it will not be possible anymore.

Reply:
OK, I hadn't understood this design choice